tests/models/test_tensorflow.py

```diff
@@ -131,6 +134,33 @@ def test_save_weights(tf_model, tmp_path):
     assert_models_equal(tf_model, loaded_model)
 
 
+@pytest.mark.skipif(not _is_tensorflow_using_keras3_api(), reason="Testing behaviour for Keras >= 3.0")
```

Comment:
The tests above were asserting that the destination was a directory, so I opted to keep them as is and run them conditionally based on the tensorflow version. Adding new tests here to run conditionally based on the tensorflow version and test the new API